### PR TITLE
Improve pending tx modal + shell/download file error handling

### PR DESCRIPTION
**Pull request by Redm4x**
- Added an FAQ entry explaining why UP arrow and TAB autocompletion may not work (using sh shell)
- Added links to FAQ when trying tu use up arrow in sh shell and when shell is not connecting
- Fixed a bug with file download and improved error handling
- Made the "Pending tx" modal unclosable while keplr/leap popup is open #29

### CODE REVIEW

File: deploy-web/src/components/deploymentDetail/DeploymentLeaseShell.tsx [hunk 1]
```diff
@@ -268,10 +277,24 @@ export const DeploymentLeaseShell: React.FunctionComponent<Props> = ({ leases })
                 )}
               </Box>
 
+              {showArrowAndTabWarning && (
+                <Alert variant="standard" severity="warning" sx={{ borderRadius: 0, marginBottom: 1 }}>
+                  <Link href="/faq#shell-arrows-and-completion" target="_blank" style={{ display: "inline-flex", alignItems: "center" }}>
```
Comment (baktun14) [inline deploy-web/src/components/deploymentDetail/DeploymentLeaseShell.tsx]: `UrlService`

File: deploy-web/src/components/deploymentDetail/DeploymentLeaseShell.tsx [hunk 2]
```diff
@@ -268,10 +277,24 @@ export const DeploymentLeaseShell: React.FunctionComponent<Props> = ({ leases })
...
+                  </Link>
+                </Alert>
+              )}
+
               <ViewPanel stickToBottom style={{ overflow: "hidden" }}>
                 {isConnectionClosed && (
                   <Alert variant="standard" severity="warning" sx={{ borderRadius: 0 }}>
-                    The connection to your Cloudmos Shell was lost.
+                    The connection to your Cloudmos Shell was lost. (
+                    <Link href="/faq#shell-lost" target="_blank" style={{ display: "inline-flex", alignItems: "center" }}>
```
Comment (baktun14) [inline deploy-web/src/components/deploymentDetail/DeploymentLeaseShell.tsx]: `UrlService`

File: deploy-web/src/components/layout/TransactionModal.tsx [hunk 3]
```diff
@@ -3,18 +3,19 @@ import { Popup } from "../shared/Popup";
...
+export const TransactionModal: React.FunctionComponent<Props> = ({ state, open, onClose }) => {
   return (
     <Popup
       fullWidth
       open={open}
       variant="custom"
-      title={<>Transaction Pending</>}
+      title={state === "waitingForApproval" ? <>Waiting for tx approval</> : <>Transaction Pending</>}
       actions={[]}
       onClose={onClose}
```
Comment (baktun14) [inline deploy-web/src/components/layout/TransactionModal.tsx:20]: Is the X on the popup still visible?

Comment (Redm4x) [reply deploy-web/src/components/layout/TransactionModal.tsx:20]: Nope it's hidden when `onClose` is `undefined`

Comment (baktun14) [reply deploy-web/src/components/layout/TransactionModal.tsx:20]: Ok great

**Review by baktun14 — APPROVED** [review]

**Pull request merged**